pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_add_column.go

```diff
@@ -174,8 +176,31 @@ func alterTableAddColumn(
 		))
 	}
 	if desc.IsComputed() {
+
 		expr := b.ComputedColumnExpression(tbl, d)
 		spec.colType.ComputeExpr = b.WrapExpression(tbl.TableID, expr)
+
+		// run the same validation in sql/row/row_converter.go GenerateInsertRow function
+		var txCtx transform.ExprTransformContext
+		// compute expression can have dependency to another column,
+		// in this case line below will throw an error and validation is not necessary
+		typedExpr, err := tree.TypeCheck(b, expr, b.SemaCtx(), spec.colType.Type)
```

Comment:
We also need to fix the same issue in `CREATE TABLE`. I suggest performing these checks in  https://github.com/cockroachdb/cockroach/blob/fc1fa24f38a0b4f1e60bd079f231de37a2e842bd/pkg/sql/catalog/schemaexpr/computed_column.go#L45 which should be utilized by both ALTER TABLE and CREATE TABLE.